### PR TITLE
Add feedback messages provided by Stanford personnel.

### PR DESCRIPTION
**Pull request by jsewall**
For maintaining these .brd files, use the CTAT mass production files in OLI.

### CODE REVIEW

File: stattutor/public/js/stattutor.js [hunk 1]
```diff
@@ -647,6 +647,37 @@ function gen_k_of_n_checkbox_feedback(exp, act, intro, studentTerms, itemName, i
...
+ * @param {Array<string>} studentTests student-friendly names for each statistical test
+ * @param {Array<string>} descTests descriptions of when to use each test
+ * @param {Array<string>} useTests short statement of the use of each test
+ * @param {string} intro prefix to all responses
+ * @param {string} finish suffix to all responses
+ * @return {string} feedback message
+ */
+function gen_1_of_n_stat_test_feedback(act, tests, studentTests, descTests, useTests, intro, finish) {
+	var i=0;
+	while(i<tests.length && act != tests[i]) ++i;
```
Comment (stvstnfrd) [inline stattutor/public/js/stattutor.js:662]: Not a deal-breaker, but a note for future reference:
We're reinventing the wheel here;
JavaScript provides this functionality by default,
via the `Array.prototype.indexOf` method [1].

```javascript
var index = tests.indexOf(act);
if(index === -1) {
    // item not found
    return intro;
} else {
    // item found
    return intro+" "+studentTests[i]+" is used when you "+descTests[i]+" We are <b>not</b> testing "+useTests[i]+" "+finish;
}
```

[1] https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf

File: stattutor/public/js/stattutor.js [hunk 2]
```diff
@@ -647,6 +647,37 @@ function gen_k_of_n_checkbox_feedback(exp, act, intro, studentTerms, itemName, i
...
+ * Generate a feedback message for a choose-one-of-N question.
+ * @param {string} act student's answer
+ * @param {Array<string>} answers all inputs available to the student; act must be one of these
+ * @param {Array<string>} msgs feedback message for each element in answers
+ * @param {string} defaultMsg return this if nothing matches
+ * @return {string} feedback message
+ */
+function gen_1_of_n_feedback(act, answers, msgs, defaultMsg) {
+	var i=0;
+	while(i<answers.length && act != answers[i]) ++i;
```
Comment (stvstnfrd) [inline stattutor/public/js/stattutor.js:676]: See note above about using JavaScript's standard library.

File: stattutor/public/js/stattutor.js [hunk 3]
```diff
@@ -647,6 +647,37 @@ function gen_k_of_n_checkbox_feedback(exp, act, intro, studentTerms, itemName, i
...
+ * @param {string} act student's answer
+ * @param {Array<string>} tests all answers available; act must be one of these
+ * @param {Array<string>} studentTests student-friendly names for each statistical test
+ * @param {Array<string>} descTests descriptions of when to use each test
+ * @param {Array<string>} useTests short statement of the use of each test
+ * @param {string} intro prefix to all responses
+ * @param {string} finish suffix to all responses
+ * @return {string} feedback message
+ */
+function gen_1_of_n_stat_test_feedback(act, tests, studentTests, descTests, useTests, intro, finish) {
```
Comment (stvstnfrd) [inline stattutor/public/js/stattutor.js:660]: I'm not seeing this function called w/in the JavaScript;
it looks like it's referenced in the BRD file though?

**Review by stvstnfrd — APPROVED** [review]
I left a note about a better method for performing array lookups, but I'm okay enough with this for now.

@dcadams Do you mind being a second set of eyes?

File: stattutor/public/js/stattutor.js (hunk 3 above)
Comment (jsewall) [reply stattutor/public/js/stattutor.js:660]: Yes, each of these new JavaScript functions is called from within the .brd files to generate formulaic feedback messages.
```
$ cd stattutor/public/problem_files
$ grep -l gen_1_of_n_stat_test_feedback $(find m[0-9]_* -name "*.brd")
m7_auto_premiums/auto_premiums.brd
$ grep -l gen_1_of_n_feedback $(find m[0-9]_* -name "*.brd")
m4_cell_phones/cell_phones.brd
m4_cell_phones_1q/cell_phones_1q.brd
m5_extra_credit/extra_credit.brd
m5_extra_credit_2q/extra_credit_2q.brd
m6_low_birth_weight/low_birth_weight.brd
m7_auto_premiums/auto_premiums.brd
```

File: stattutor/public/js/stattutor.js (hunk 1 above)
Comment (jsewall) [reply stattutor/public/js/stattutor.js:662]: Thanks for the tip.

**Comment by dcadams** [comment]
:+1:

**Pull request merged**